tests/build/chains.go

```diff
@@ -159,44 +161,70 @@ var _ = framework.ChainsSuiteDescribe("Tekton Chains E2E tests", func() {
-				tr, err := kubeController.RunTask(taskGenerator, taskTimeout)
+				pr, err := kubeController.RunPipeline(generator, pipelineRunTimeout)
 				Expect(err).NotTo(HaveOccurred())
-				Expect(kubeController.WatchTaskPod(tr.Name, taskTimeout)).To(Succeed())
+				Expect(kubeController.WatchPipelineRun(pr.Name, pipelineRunTimeout)).To(Succeed())
 
-				// Refresh our copy of the TaskRun for latest results
-				tr, err = kubeController.Tektonctrl.GetTaskRun(tr.Name, tr.Namespace)
+				// Refresh our copy of the PipelineRun for latest results
+				pr, err = kubeController.Tektonctrl.GetPipelineRun(pr.Name, pr.Namespace)
 				Expect(err).NotTo(HaveOccurred())
-				Expect(tr.Status.TaskRunResults).To(Equal([]v1beta1.TaskRunResult{
-					{Name: "OUTPUT", Value: "[]\n"},
-					{Name: "PASSED", Value: "true\n"},
-				}))
+				tr, err := kubeController.GetTaskRunStatus(pr, "verify-enterprise-contract")
+				Expect(err).NotTo(HaveOccurred())
+				Expect(tr.Status.TaskRunResults).Should(ContainElements(
+					tekton.MatchTaskRunResultWithJSONValue("OUTPUT", `[
+						{
+							"filename": "/shared/ec-work-dir/input/input.json",
+							"namespace": "main",
+							"successes": 1
+						}
+					]`),
+					tekton.MatchTaskRunResult("PASSED", "true"),
+				))
 			})
 
 			g.It("does not pass when tests are not satisfied on non-strict mode", func() {
-				taskGenerator.StrictPolicy = "0"
-				tr, err := kubeController.RunTask(taskGenerator, taskTimeout)
+				generator.StrictPolicy = "0"
+				pr, err := kubeController.RunPipeline(generator, pipelineRunTimeout)
 				Expect(err).NotTo(HaveOccurred())
-				Expect(kubeController.WatchTaskPod(tr.Name, taskTimeout)).To(Succeed())
+				Expect(kubeController.WatchPipelineRun(pr.Name, pipelineRunTimeout)).To(Succeed())
 
-				// Refresh our copy of the TaskRun for latest results
-				tr, err = kubeController.Tektonctrl.GetTaskRun(tr.Name, tr.Namespace)
+				// Refresh our copy of the PipelineRun for latest results
+				pr, err = kubeController.Tektonctrl.GetPipelineRun(pr.Name, pr.Namespace)
+				Expect(err).NotTo(HaveOccurred())
+				tr, err := kubeController.GetTaskRunStatus(pr, "verify-enterprise-contract")
 				Expect(err).NotTo(HaveOccurred())
-				Expect(tr.Status.TaskRunResults).To(Equal([]v1beta1.TaskRunResult{
-					{Name: "OUTPUT", Value: "[\n  {\n    \"code\": \"test_data_empty\",\n    \"msg\": \"Empty test data provided\"\n  }\n]\n"},
-					{Name: "PASSED", Value: "false\n"},
-				}))
+				Expect(tr.Status.TaskRunResults).Should(ContainElements(
+					tekton.MatchTaskRunResultWithJSONValue("OUTPUT", `[
+						{
+							"filename": "/shared/ec-work-dir/input/input.json",
+							"namespace": "main",
+							"successes": 0,
+							"failures": [
+								{
+									"msg": "Empty test data provided",
+									"metadata": {
+										"code": "test_data_empty"
+									}					
+								}
+							]
+						}
+					]`),
+					tekton.MatchTaskRunResult("PASSED", "false"),
```

Comment:
Grew really tired of putting JSON in a string in the exact same format it comes out in the TaskRun result, so I created this matcher. The downside of using it is that `ContainElements` doesn't use `FailureMessage` so the failures are not nicely presented.

The matcher compares as JSON regardless of insignificant whitespace (i.e. formatting) in `MatchTaskRunResultWithJSONValue` and trims the value in string comparison in `MatchTaskRunResult`.